moveit_core/planning_request_adapter/include/moveit/planning_request_adapter/planning_request_adapter.h

```diff
@@ -72,7 +74,11 @@ class PlanningRequestAdapter
     std::string adapter_name = typeid(*this).name();
     // Try to demangle the name
     int status;
+#if defined(_MSC_VER)
+    std::string demangled_name = adapter_name;
+#else
     std::string demangled_name = abi::__cxa_demangle(adapter_name.c_str(), NULL, NULL, &status);
+#endif
```

Comment:
Not necessarily something that needs to be addressed in this pull-request, but I would vote for removing this ugly chunk altogether and making the `initialize` method pure virtual instead.

Any comments by other maintainers?

Reply:
I agree that this is ugly and should probably be at least abstracted away, we could have a general demangle.h to get these names that'd know how to support different platform on it's own but yes, the best would be to remove hacky stuff like this altogether.

What if we drop the `adapter_name` variable and define a `std::string getAdapterName() = 0;`? This would enforce that whoever implement an adapter defines this function to return something. May as well return a correct name. Separate PR though, let's not hold this one up for that.

Reply:
@bmagyar The logic above will always trigger when `initialize` is not overridden by the child class.
It's basically a run-time, warning-only version of a virtual `initialize` function.
There is no other place where the adapter name needed in the base class.

So instead of adding a *new* pure virtual method, as you proposed, I proposed to enforce the warning condition at compile time by setting `virtual void initialize(...) = 0`.